test/games/strategy/triplea/delegate/BattleCalculatorTest.java

```diff
@@ -139,18 +144,24 @@ public void testAACasualtiesLowLuckMixedWithChooseAACasualties() {
     final Collection<Unit> planes = bomber(data).create(6, british(data));
     planes.addAll(fighter(data).create(6, british(data)));
     final Collection<Unit> defendingAA = territory("Germany", data).getUnits().getMatches(Matches.UnitIsAAforAnything);
-    m_bridge.setRemote(new DummyTripleAPlayer() {
-      @Override
-      public CasualtyDetails selectCasualties(final Collection<Unit> selectFrom,
-          final Map<Unit, Collection<Unit>> dependents, final int count, final String message, final DiceRoll dice,
-          final PlayerID hit, final Collection<Unit> friendlyUnits, final PlayerID enemyPlayer,
-          final Collection<Unit> enemyUnits, final boolean amphibious, final Collection<Unit> amphibiousLandAttackers,
-          final CasualtyList defaultCasualties, final GUID battleID, final Territory battlesite,
-          final boolean allowMultipleHitsPerUnit) {
-        final List<Unit> selected = Match.getNMatches(selectFrom, count, Matches.UnitIsStrategicBomber);
-        return new CasualtyDetails(selected, new ArrayList<>(), false);
-      }
-    });
+    TripleAPlayer player = MockObjects.getDummyPlayer();
+    when(player.selectCasualties(any(Collection.class), any(Map.class), any(int.class), any(String.class),
```

Comment:
Avoid expecting `any` object. Try to match exact values or instances. For some of these I can see it being difficult to do otherwise.